### ISSUE TITLE
Module setup cannot be finished; Results in broken Shop (v2.4.0)

### ISSUE THREAD

**Issue by kevingimbel** [Smile-SA/elasticsuite#698]
### Preconditions

AWS EC2 Instance (t2.micro) as Server

Magento Installed from AWS Marketplace  (https://aws.amazon.com/marketplace/pp/B071WD5L9P?qid=1516188036416&sr=0-1&ref_=srh_res_product_title)

Magento Version : Magento CE 2.2

ElasticSuite Version : 2.4.0

Environment : Developer

Third party modules : 

Additional Software / Setup: 

- Ubuntu 16.04.3 LTS
- Apache 2.4.18
- MySQL 5.7.20
- PHP 7.0.22
- FPM and Opcache enabled
- SSL via Lets Encrypt

### Steps to reproduce

1. Install clean Magento
2. Install ElasticSuite with composer
3. Run `bin/magento setup:upgrade`

### Expected result

1. The module is installed and sets up its database tables

### Actual result

First I got an "undefined offset" issue:
 
```txt 
Notice: Undefined offset: 1 in /var/www/magento/vendor/smile/elasticsuite/src/module-elasticsuite-core/Client/ClientFactory.php on line 95
```

Then I tried to clear caches and re-run the setup because in my experience that sometimes fixes issues in Magento. On the second run I got the following error:

```txt
$ sudo -u www-data php bin/magento setup:upgrade
...
Module 'Smile_ElasticsuiteCore':
Module 'Smile_ElasticsuiteCatalog':
Installing data... No alive nodes found in your cluster
```

The Magento Front-End (and Backend) now shows an Error Report page ("There has been an error processing your request"). The report contains the following:

```
{
  "0": "Please upgrade your database: Run \"bin\/magento setup:upgrade\" from the Magento root directory.\nThe following modules are outdated:\nSmile_ElasticsuiteCatalog data: current version - none, required version - 1.2.2\nSmile_ElasticsuiteCatalogRule data: current version - none, required version - 1.0.0\nSmile_ElasticsuiteCatalogOptimizer data: current version - none, required version - 1.0.1\nSmile_ElasticsuiteSwatches data: current version - none, required version - 0.0.1\nSmile_ElasticsuiteThesaurus data: current version - none, required version - 1.0.0\nSmile_ElasticsuiteTracker data: current version - none, required version - 1.0.0\nSmile_ElasticsuiteVirtualCategory data: current version - none, required version - 1.1.0",
  "1": "#0 \/var\/www\/magento\/vendor\/magento\/framework\/Interception\/Interceptor.php(121): Magento\\Framework\\Module\\Plugin\\DbStatusValidator->beforeDispatch(Object(Magento\\Framework\\App\\FrontController\\Interceptor), Object(Magento\\Framework\\App\\Request\\Http))\n#1 \/var\/www\/magento\/vendor\/magento\/framework\/Interception\/Interceptor.php(153): Magento\\Framework\\App\\FrontController\\Interceptor->Magento\\Framework\\Interception\\{closure}(Object(Magento\\Framework\\App\\Request\\Http))\n#2 \/var\/www\/magento\/generated\/code\/Magento\/Framework\/App\/FrontController\/Interceptor.php(26): Magento\\Framework\\App\\FrontController\\Interceptor->___callPlugins('dispatch', Array, Array)\n#3 \/var\/www\/magento\/vendor\/magento\/framework\/App\/Http.php(135): Magento\\Framework\\App\\FrontController\\Interceptor->dispatch(Object(Magento\\Framework\\App\\Request\\Http))\n#4 \/var\/www\/magento\/vendor\/magento\/framework\/App\/Bootstrap.php(256): Magento\\Framework\\App\\Http->launch()\n#5 \/var\/www\/magento\/index.php(39): Magento\\Framework\\App\\Bootstrap->run(Object(Magento\\Framework\\App\\Http))\n#6 {main}",
  "url": "xxxx",
  "script_name": "\/index.php"
}

```
`xxxx` in the above log was the Admin URL.

When I disable all the ElasticSuite module(s) with the following command everything is working again.

```sh
$ sudo -u www-data php bin/magento module:disable Smile_ElasticsuiteCore Smile_ElasticsuiteCatalog Smile_ElasticsuiteSwatches Smile_ElasticsuiteCatalogRule Smile_ElasticsuiteVirtualCategory Smile_ElasticsuiteThesaurus Smile_ElasticsuiteCatalogOptimizer Smile_ElasticsuiteTracker
```

**Comment by minhluan259** [comment]
I think you're missing server config https://github.com/Smile-SA/elasticsuite/wiki/ServerConfig

**Comment by kevingimbel** [comment]
@minhluan259 How can I configure an Elasticsearch server when the module isn't setup? My impression was that I first install the module and then use the Backend settings to point it to my Elasticsearch installation (which is on another AWS Instance). 

Does the module not work if Elasticsearch is on a different server?

**Comment by afoucret** [comment]
I agree with the two previous comments.

You can have a look at this issue : #692
You can setup your environnement using env variables as explained in one of my response.

In the next release we will add additional params to the setup:install command to allow es configuration to be done in the same way as MySQL or Redis.

Closed for being a duplicate of #692

**Comment by surendiranganesan** [comment]
Hi @afoucret ,

I too facing the same issue. I installed in magento2.2.2 in my localhost wamp and installed elastic suite 2.5.0 successfully.

Affter installation when i try to open both front end and back end i am facing the same issue

Please upgrade your database: Run "bin/magento setup:upgrade" from the Magento root directory.
The following modules are outdated:
```
Smile_ElasticsuiteCatalog data: current version - none, required version - 1.4.0
Smile_ElasticsuiteCatalogRule data: current version - none, required version - 1.0.0
Smile_ElasticsuiteCatalogOptimizer data: current version - none, required version - 1.0.1
Smile_ElasticsuiteSwatches data: current version - none, required version - 0.0.1
Smile_ElasticsuiteThesaurus data: current version - none, required version - 1.0.0
Smile_ElasticsuiteTracker data: current version - none, required version - 1.1.0
Smile_ElasticsuiteVirtualCategory data: current version - none, required version - 1.1.0
```
you hve shared a link []( https://github.com/Smile-SA/elasticsuite/issues/692). i am new to magento i dont know exactly what to do. can you please help me.
